### ISSUE TITLE
iOS swipe back flashes last view for a brief moment.

### ISSUE THREAD

**Issue by allantatter** [reactjs/react-router#3095]
Hello

I have a website built on React, Redux and React-Router. I have an issue with iOS device, when going back with a swipe left, then during the swipe, I see the previous view as I should and when I release my thumb from the screen then for a short time there is the previous page that you cannot interact with. Then, the JS code recovers and flashes the next view for a brief moment and then switches to the previous view. Is it even possible to get rid of that flash? Is it better if a make a video of my phone when reproducing the issue?

**Comment by posabsolute** [comment]
Ever fixed your issue? I got the same problem, it seems that for some reasons the page scroll to top on ios before the view change, does not happen on chrome.

**Comment by allantatter** [comment]
I did not, but it seems that when listening directly for `popstate` event and then doing smth by yourself w/o React/Router then somehow maybe it can be solved (kind of).

So in componentDidMount you would have:

```
    window.onpopstate = () => {
      this.refs.container.style.display = 'none';
    };
```

And the highest level element in your React component you would have `ref="container"`. In this case at least my last view is hidden and not flashed to the user.

About the scroll issue, the thing is that the browser tries to scroll by itself to the correct position and maybe in your case (as in mine) I don't want that functionality and want to specify the scroll position by myself. There is a feature named [scrollRestoration](https://developer.mozilla.org/en/docs/Web/API/History) for that but it's not implemented on iOS/Android yet (and no hope in the near future) so I chose to use a [polyfill](https://github.com/bfred-it/scroll-restoration-polyfill) for that but it's far from ideal.
